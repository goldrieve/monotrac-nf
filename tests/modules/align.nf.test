nextflow_process {

    name "Test Process ALIGN"
    script "main.nf"
    process "ALIGN"

    test("Should run alignment") {

        when {
            process {
                """
                input[0] = file("${projectDir}/tests/data/SVP.fas")
                input[1] = [
                    file("${projectDir}/data/isolate_fasta/108AT.fas"),
                    file("${projectDir}/data/isolate_fasta/108BT.fas"),
                    file("${projectDir}/data/isolate_fasta/15BT-relapse.fas")
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}