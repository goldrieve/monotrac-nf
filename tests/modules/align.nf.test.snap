{
    "Should run alignment": {
        "content": [
            {
                "0": [
                    "aligned.fas:md5,860dedda87c4b1bb78f081ac1ebf5273"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T11:04:01.717546648"
    }
}